### PR TITLE
Implement Analyze Stacktrace

### PR DESCRIPTION
**Pull request by ayoub-benali**
closes https://github.com/scalameta/metals-sublime/issues/52

Peek 2021-11-28 20-22

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]
LGTM! We can also do the fix for the supported `command link`  first and use the HTML output

**Comment by ayoub-benali** [comment]
@tgodzik I pushed an implementation change using HTML output and regex to rewrite the link.

Peek 2021-11-29 12-17

File: st4/__init__.py [hunk 1]
```diff
@@ -116,6 +121,16 @@ def m_metals_executeClientCommand(self, params: Any) -> None:
...
                         flags=sublime.ENCODED_POSITION
                     )
+        if params.get('command') == 'metals-show-stacktrace':
+            args = params.get('arguments')
+            def _replace_link(match) -> str:
+                url = urlparse(match.group(1))
+                command_args = json.loads(unquote(url.query))
+                path = urlparse(command_args[0]).path
+                location = command_args[1] + 1
+                return "href='subl:open_file_encoded {{\"file\": \"{}:{}\"}}'".format(path, location)
```
Comment (tgodzik) [inline st4/__init__.py:131]: We can merge as is and maybe update the support in Metals later (with sublime/vscode specific commands)

Comment (ayoub-benali) [reply st4/__init__.py:131]: Yes I will create an issue on Metals and make a PR for specific commands

**Pull request merged**